contrib/pkg/awsprivatelink/enable.go

```diff
@@ -0,0 +1,217 @@
+		log.WithError(err).Fatal("Failed to get HiveConfig/hive")
+	}
+	if hiveConfig.Spec.AWSPrivateLink != nil {
+		log.Fatal("AWS Private Link is already enabled. If a previous configuration attempt did not complete, " +
+			"you can clean up via `hiveutil awsprivatelink disable` and try again.")
+	}
+
+	// Get active cluster's VPC, filtering by infra-id
+	describeVPCsOutput, err := o.awsClients.DescribeVpcs(&ec2.DescribeVpcsInput{
+		Filters: []*ec2.Filter{
+			{
+				Name:   aws.String("tag-key"),
+				Values: []*string{aws.String("kubernetes.io/cluster/" + o.infraId)},
+			},
+		},
+	})
+	if err != nil {
+		log.WithError(err).Fatal("Failed to get VPC of the active cluster")
+	}
+	if len(describeVPCsOutput.Vpcs) == 0 {
+		log.Fatal("VPC of the active cluster not found")
+	}
+	vpcID := *describeVPCsOutput.Vpcs[0].VpcId
+	log.Debugf("Found VPC ID = %v for the active cluster", vpcID)
+
+	// Create Hub account credentials Secret
+	hiveNS := operatorutils.GetHiveNamespace(hiveConfig)
+	hubCredentialsSecret, err := o.generateAWSCredentialsSecret(hiveNS)
+	if err != nil {
+		log.WithError(err).Fatal("Failed to generate Secret with AWS credentials")
+	}
+	switch err = o.dynamicClient.Create(context.Background(), hubCredentialsSecret); {
+	case err == nil:
+		log.Infof("Secret/%s created in namespace %s", awsutils.PrivateLinkHubAcctCredsName, hiveNS)
+	case apierrors.IsAlreadyExists(err):
+		log.Warnf("Secret/%s already exists in namespace %s", awsutils.PrivateLinkHubAcctCredsName, hiveNS)
+	default:
+		log.WithError(err).Fatalf("Failed to create Secret/%s in namespace %s", awsutils.PrivateLinkHubAcctCredsName, hiveNS)
+	}
+
+	// Update HiveConfig
+	hiveConfig.Spec.AWSPrivateLink = &hivev1.AWSPrivateLinkConfig{
+		AssociatedVPCs: []hivev1.AWSAssociatedVPC{
+			{
+				AWSPrivateLinkVPC: hivev1.AWSPrivateLinkVPC{VPCID: vpcID, Region: o.region},
+			},
+		},
+		CredentialsSecretRef: corev1.LocalObjectReference{Name: hubCredentialsSecret.Name},
+		DNSRecordType:        hivev1.AWSPrivateLinkDNSRecordType(o.dnsRecordType),
+	}
+	if err = o.dynamicClient.Update(context.Background(), hiveConfig); err != nil {
+		log.WithError(err).Fatal("Failed to update HiveConfig")
+	}
+	log.Info("HiveConfig updated")
+
+	return nil
+}
+
+func (o *enableOptions) getRegionAndInfraId() (string, string, error) {
+	infrastructure := &configv1.Infrastructure{}
+	if err := o.dynamicClient.Get(context.Background(), types.NamespacedName{Name: "cluster"}, infrastructure); err != nil {
+		return "", "", err
+	}
+	if infrastructure.Status.PlatformStatus == nil {
+		return "", "", errors.New("Infrastructure.status.platformStatus is empty")
+	}
+	if infrastructure.Status.PlatformStatus.AWS == nil {
+		return "", "", errors.New("Infrastructure.status.platformStatus.aws is empty")
+	}
+
+	return infrastructure.Status.PlatformStatus.AWS.Region, infrastructure.Status.InfrastructureName, nil
+}
+
+// Adapted from contrib/pkg/adm/managedns/enable.go:generateAWSCredentialsSecret()
```

Comment:
The function I'm referencing is different in some ways, e.g. the Secret generated has a randomized name, and no label. It also has a different receiver (it is a method of a different type). So I guess it could be complicated to use it.

Reply:
Thanks for taking the time to look into this. Would be great to refactor in future PRs.